### ISSUE TITLE
[Debugger] coreclr-debug project.json should be updated for dotnet rc2 schema.

### ISSUE THREAD

**Issue by MatthewDavidCampbell** [OmniSharp/omnisharp-vscode#374]
Using 1.0.11 of csharp...

The default tasks.json from a project initialized with DotNet (latest preview version RC2) uses the DotNet command.  This fails to build the **coreclr-debug** **project.json** file which has directives associated with the old tool chain (i.e. compilationOptions rather than buildOptions).  Thought I could just update the project.json file but it is reloaded on startup (of Visual Studio Core).

Is this a bug or it there a way to tell Code for extensions to use the older toolchain?

**Comment by DustinCampbell** [comment]
The extension is intended to work with .NET Core RC2 tool chain, and not the older one. The coreclr-debug project.json isn't actually intended to be built. It's there for dotnet restore.

**Comment by chuckries** [comment]
@MatthewDavidCampbell I am a little confused about the issue you are actually raising. What failure(s) are you seeing and can you provide an error log or screenshot?

**Comment by DustinCampbell** [comment]
@chuckries: I think the concern is that the project.json that we generate to restore the debugger uses the old project.json schema.

https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/coreclr-debug/main.ts#L327

**Comment by chuckries** [comment]
I understand and we should update the file. I haven't personally seen this cause failures with dotnet RC2 though, and was hoping to get an error log if available.

**Comment by MatthewDavidCampbell** [comment]
Attached a log of what happens after installing the extension (see below).  To recreate the issue is easy.  Create a new project with **dotnet new** (note: version is 1.0.0-preview1-002702), restore it and run with dotnet cli.  Then opened that folder in VS Code.  Installed the csharp extension again.  Launching the console task also yields again.  Tried editing the project.json file for the debugger manually but it gets pulled again when building the dummy project.  Thus the question I raised why the build of the extension is tied to the build of the project (i.e. they are activated both with dotnet).  Also, just popped only VSCode today so still trying to learn the tool.

[PostInstallOfCSharpExtension.txt](https://github.com/OmniSharp/omnisharp-vscode/files/282960/PostInstallOfCSharpExtension.txt)

**Comment by chuckries** [comment]
I don't believe the issue you are seeing is related to compilationOptions <-> buildOptions. We need to fix that but it is not the cause of the error you are seeing.

After the C# extension is installed to Visual Studio Code, it will install the C# debugger autoamatically as a secondary step. The extension installs the debugger using the following steps:
1. The extension generates a project.json file in `.vscode\extensions\ms-vscode.csharp-1.0.11\coreclr-debug`
2. The extension runs `dotnet restore` and `dotnet publish` on this project.json. This causes the debugger binaries to be downloaded and copied to `.vscode\extensions\ms-vscode.csharp-1.0.11\coreclr-debug\debugAdapters`
3. As part of this restore and publish, an executable is created called dummy.exe
4. The extension renames dummy.exe to OpenDebugAD7.exe. When run, OpenDebugAD7.exe will run OpenDebugAD7.dll, which starts the debugger.

According to your log, something is going wrong with this rename step:
```
Error: The process cannot access the file 'C:\Users\MYO\.vscode\extensions\ms-vscode.csharp-1.0.11\coreclr-debug\debugAdapters\dummy.exe' because it is being used by another process.

System.IO.IOException: The process cannot access the file 'C:\Users\MYO\.vscode\extensions\ms-vscode.csharp-1.0.11\coreclr-debug\debugAdapters\dummy.exe' because it is being used by another process.
   at System.IO.Win32FileSystem.OpenHandle(String fullPath, Boolean asDirectory)
   at System.IO.Win32FileSystem.SetLastWriteTimeInternal(String fullPath, DateTimeOffset time, Boolean asDirectory)
   at System.IO.Win32FileSystem.SetLastWriteTime(String fullPath, DateTimeOffset time, Boolean asDirectory)
   at System.IO.File.SetLastWriteTimeUtc(String path, DateTime lastWriteTimeUtc)
   at Microsoft.DotNet.Cli.Utils.CoreHost.CopyTo(String destinationPath, String hostExeName)
   at Microsoft.DotNet.Tools.Publish.PublishCommand.RenamePublishedHost(ProjectContext context, String outputPath, CommonCompilerOptions compilationOptions)
   at Microsoft.DotNet.Tools.Publish.PublishCommand.PublishProjectContext(ProjectContext context, String buildBasePath, String outputPath, String configuration, Boolean nativeSubdirectories)
   at Microsoft.DotNet.Tools.Publish.PublishCommand.PublishAllProjects()
   at Microsoft.DotNet.Tools.Publish.PublishCommand.<>c__DisplayClass0_0.<Run>b__0()
   at Microsoft.DotNet.Cli.CommandLine.CommandLineApplication.Execute(String[] args)
   at Microsoft.DotNet.Tools.Publish.PublishCommand.Run(String[] args)
dotnet exited with error code 1
```

You should not be manually changing `.vscode\extensions\ms-vscode.csharp-1.0.11\coreclr-debug\project.json`. It is generated by the extension.

The tasks.json and launch.json files that are generated for you by the extension have nothing to do with the project.json in the debuggers extension directory. 

I suggest you do the following:
1. Manually delete the entire `.vscode\extensions\ms-vscode.csharp-1.0.11` directory.
2. Restart your machine.
3. Reinstall the C# extension.

**Comment by MatthewDavidCampbell** [comment]
Did those steps except for the restart earlier. Tried unstalling first but saw that the extension wasn't completely deleted just ignored. After a clean and then install the issue persists. 

Thought the blocked by another process might have been due to a failed build of the debugger. 

What decides what command builds an extension? Hardcoded for **dotnet** in the PATH? Wild stab that the build for the project and the extension would tied since it was debugger related.

**Comment by chuckries** [comment]
Nothing "builds an extension". These dotnet steps are specific to our extension and can be seen [here](https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/coreclr-debug/main.ts#L58). This code is run on extension "activiation". The C# extension is "activated" when you open a .cs file or a directory containing a project.json at the root.

**Comment by DustinCampbell** [comment]
@MatthewDavidCampbell, I'm confused too. What do you mean by, "builds an extension"? You don't have to build the C# extension or run **dotnet** on the extension itself. Perhaps you could provide some screenshots or precise steps to help us better understand what's happening?

**Comment by MatthewDavidCampbell** [comment]
Same version.  Same process.  And this time around no error.  Machine has been restarted.  Before, I had created a dummy project (via **dotnet new**) called _hwapp_.  When testing the csharp extension the project hwapp was created a while ago with an older version of the dotnet cli.  I deleted it and recreated with the latest preview version.  This evening, I tested first by creating another project (i.e. another directory/folder) called _helloworld_ and everything worked fine (as is shown in the attached files).  Redid but picked the old name _hwapp_ and it worked.

No idea what happened other than a restart.  Something must have been cached.  What and where would be cool to know.


[DotNetCliNewProject.txt](https://github.com/OmniSharp/omnisharp-vscode/files/285242/DotNetCliNewProject.txt)

![helloworldpostrestart](https://cloud.githubusercontent.com/assets/1491747/15588264/46df1176-238d-11e6-9574-c1b197a06f8c.png)
![helloworldrestart](https://cloud.githubusercontent.com/assets/1491747/15588265/46e03006-238d-11e6-9540-381d3b197112.png)
![helloworldpreinstall](https://cloud.githubusercontent.com/assets/1491747/15588263/46dcd5be-238d-11e6-8374-f03d3edf7425.png)
